### PR TITLE
Unbelievable work by hujh14. 

### PR DESCRIPTION
**Pull request by Vladkryvoruchko**
omg, man - you are so sick!
Cant believe how much work you have done.
Can you please describe changes?

### CODE REVIEW

**Comment by hujh14** [comment]
There's a bunch of unrelated stuff where I'm trying to add a GAN to improve the results.

I can break out just the pspnet to keras portion. I think I worked out most of the issues. I know at least my feedforward is identical to pycaffe because I printed the activations for each layer and they are within 1e-5 of eachother for all values.

**Comment by hujh14** [comment]
Training or rather finetuning is more difficult. The key to getting useable results is not to touch the batch normalization layers. For some reason you can't freeze the batch norm layers in Keras so I had to change some lines in my version.

**Comment by hujh14** [comment]
I'll push a working feedforward copy today or tomorrow.
